 # Get started with TensorFlow 2.0 for beginners
 In this example we train an image classifier.

 Load packages

In [21]:
from __future__ import absolute_import, division, print_function, unicode_literals
import tensorflow as tf
print("Tensorflow version: " + tf.__version__)



Tensorflow version: 2.0.0-rc0


 Verify GPU

In [22]:
device_lib.list_local_devices()

[name: "/device:CPU:0"
 device_type: "CPU"
 memory_limit: 268435456
 locality {
 }
 incarnation: 2509953640761771831, name: "/device:XLA_CPU:0"
 device_type: "XLA_CPU"
 memory_limit: 17179869184
 locality {
 }
 incarnation: 16297973117327656317
 physical_device_desc: "device: XLA_CPU device", name: "/device:XLA_GPU:0"
 device_type: "XLA_GPU"
 memory_limit: 17179869184
 locality {
 }
 incarnation: 10881540536915600794
 physical_device_desc: "device: XLA_GPU device", name: "/device:GPU:0"
 device_type: "GPU"
 memory_limit: 1424490496
 locality {
   bus_id: 1
   links {
   }
 }
 incarnation: 12042585665249375850
 physical_device_desc: "device: 0, name: GeForce MX150, pci bus id: 0000:02:00.0, compute capability: 6.1"]

In [23]:
tf.test.is_gpu_available()

True

In [24]:
tf.test.gpu_device_name()



'/device:GPU:0'

 Load and prepare the MNIST dataset. Convert the samples from integers to floating-point numbers

In [25]:
mnist = tf.keras.datasets.mnist
(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0
print("Training set shape: " + str(x_train.shape))
print("Test set shape: " + str(x_test.shape))



Training set shape: (60000, 28, 28)
Test set shape: (10000, 28, 28)


 Build the tf.keras.Sequential model by stacking layers. Choose an optimizer and loss function for training

In [26]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

In [27]:
model.summary()

Model: "sequential_3"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
flatten_3 (Flatten)          (None, 784)               0         
_________________________________________________________________
dense_6 (Dense)              (None, 128)               100480    
_________________________________________________________________
dropout_3 (Dropout)          (None, 128)               0         
_________________________________________________________________
dense_7 (Dense)              (None, 10)                1290      
Total params: 101,770
Trainable params: 101,770
Non-trainable params: 0
_________________________________________________________________


In [28]:
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])



 Train and evaluate the model

In [29]:
model.fit(x_train, y_train, epochs=5)

Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 2s 40us/sample - loss: 0.2933 - accuracy: 0.9154
Epoch 2/5
60000/60000 [==============================] - 3s 51us/sample - loss: 0.1409 - accuracy: 0.9587
Epoch 3/5
60000/60000 [==============================] - 5s 86us/sample - loss: 0.1064 - accuracy: 0.9675
Epoch 4/5
60000/60000 [==============================] - 4s 59us/sample - loss: 0.0868 - accuracy: 0.9740
Epoch 5/5
60000/60000 [==============================] - 3s 56us/sample - loss: 0.0754 - accuracy: 0.9765


In [30]:
model.evaluate(x_test, y_test)



[0.07572482982422225, 0.9762]

 The image classifier is now trained to ~98% accuracy on this dataset.